torch/csrc/inductor/aoti_torch/c/shim.h

```diff
@@ -0,0 +1,177 @@
+#ifndef AOTI_TORCH_SHIM
+#define AOTI_TORCH_SHIM
+
+#include <stddef.h>
+#include <stdint.h>
+
+// This header defines a stable C API for certain ATen functionality in
+// libtorch. The AOTInductor compiled model.so will only refer to this header
+// instead of other headers from aten/c10, which means it will NOT be able to
+// directly use any data structures or call functions from libtorch.
+//
+// What problems are we trying to solve here? If we Ahead-of-Time compile a
+// model using AOTInductor and directly call aten ops or use aten/c10 data
+// structures in the generated code, we will end up with ABI compatibility
+// breakage. By introducing a C shim layer, we can minimize the surface that
+// will cause breakage.
+
+#ifdef __GNUC__
+#define AOTI_TORCH_EXPORT __attribute__((__visibility__("default")))
+#else // !__GNUC__
+#ifdef _WIN32
+#define AOTI_TORCH_EXPORT __declspec(dllexport)
+#else // !_WIN32
+#define AOTI_TORCH_EXPORT
+#endif // _WIN32
+#endif // __GNUC__
+
+#ifdef __GNUC__
+#define AOTI_TORCH_NOINLINE __attribute__((noinline))
+#elif _MSC_VER
+#define AOTI_TORCH_NOINLINE __declspec(noinline)
+#else
+#define AOTI_TORCH_NOINLINE
+#endif
+
+struct AOTITensorOpaque {};
+using AOTITensorHandle = AOTITensorOpaque*;
+
+struct AtenTensorOpaque {};
+using AtenTensorHandle = AtenTensorOpaque*;
+
+struct AOTITensorManagerOpaque {};
+using AOTITensorManagerHandle = AOTITensorManagerOpaque*;
+
+enum AOTITorchError : int32_t {
+  Success = 0,
+  Failure = 1,
+};
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// WARNING: Change the following signatures will break ABI compatibility
```

Comment:
docs for all the functions too

Reply:
All of the functions need to document their ownership semantics (do they return borrowed or new references?)

Reply:
Added comments before functions that allocate new tensors